### ISSUE TITLE
Query for HTMLSpanElement is not working w/ getByLabelText

### ISSUE THREAD

**Issue by andrew-secret** [testing-library/dom-testing-library#499]
- `DOM Testing Library` version: `"@testing-library/react": "^10.0.0"`
- `node` version: 10
- `npm` (or `yarn`) version: 1.17.3

### Relevant code or config:

I was about to add more `a11y` to this `Icon` component which renders in the just an span element w/ an `svg` (and I know this should be in general a button 🤷‍♂️)
```tsx
<div className={styles.someClass}>
  ...
  <Icon
    label="previous month navigator"
    icon={iconCaretLeft}
    aria-disabled={props.minDate && props.minDate.getTime() >= dateForView.getTime()}
    onClick={handleClickPreviousMonth}
    className={classNames(styles.monthNavigator, {
      [styles.disabled]: props.minDate && props.minDate.getTime() >= dateForView.getTime(),
    })}
  />
  ...
</div>
```

### What you did:

I added  a `label="previous month navigator"`  to the jsx and then I wanted to test it with the `testing-library`.

```tsx
it('allows going to months before the current one', () => {
  const { getByTestId, getByText } = render(
    <DatePicker onChange={mockHandleChange} value={june} weekDays={['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']} />
  );
  fireEvent.click(getByTestId('previous month navigator'));

  expect(getByText('May 2019')).toBeInTheDocument();
});
```

### What happened:

```ts
 TestingLibraryElementError: Unable to find a label with the text of: previous month navigator

    <body>
      <div>
        <div
          class="datePicker"
        >
          <div
            class="topBar"
          >
            <span
              class="icon monthNavigator icon20"
              data-testid="previous month navigator"
              label="previous month navigator"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >


                <path
                  d="M8.828 10l3.536 3.536a1 1 0 0 1-1.414 1.414l-4.243-4.243a1 1 0 0 1 0-1.414L10.95 5.05a1 1 0 1 1 1.414 1.414L8.828 10z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />


              </svg>


            </span>
            <p
              class="typography bodySmall margin-bottom-zero margin-zero padding-zero"
            >
              June 2019
            </p>
            <span
              class="icon monthNavigator icon20"
              title="next month navigator"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >


                <path
                  d="M11.172 10L7.636 6.464A1 1 0 0 1 9.05 5.05l4.243 4.243a1 1 0 0 1 0 1.414L9.05 14.95a1 1 0 1 1-1.414-1.414L11.172 10z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />


              </svg>


            </span>
          </div>
          <div
            class="calendar"
          >
            <div
              class="weekDays"
            >
              <span
                class="day"
              >
                Mo
              </span>
              <span
                class="day"
              >
                Tu
              </span>
              <span
                class="day"
              >
                We
              </span>
              <span
                class="day"
              >
                Th
              </span>
              <span
                class="day"
              >
                Fr
              </span>
              <span
                class="day"
              >
                Sa
              </span>
              <span
                class="day"
              >
                Su
              </span>
            </div>
            <div
              class="days"
            >
              <span
                aria-disabled="true"
                class="day disabled"
              >
                27
              </span>
              <span
                aria-disabled="true"
                class="day disabled"
              >
                28
              </span>
              <span
                aria-disabled="true"
                class="day disabled"
              >
                29
              </span>
              <span
                aria-disabled="true"
                class="day disabled"
              >
                30
              </span>
              <span
                aria-disabled="true"
                class="day disabled"
              >
                31
              </span>
              <span
                class="day"
                title="1"
              >
                1
              </span>
              <span
                class="day"
                title="2"
              >
                2
              </span>
              <span
                class="day"
                title="3"
              >
                3
              </span>
              <span
                class="day"
                title="4"
              >
                4
              </span>
              <span
                class="day"
                title="5"
              >
                5
              </span>
              <span
                class="day"
                title="6"
              >
                6
              </span>
              <span
                class="day"
                title="7"
              >
                7
              </span>
              <span
                class="day"
                title="8"
              >
                8
              </span>
              <span
                class="day"
                title="9"
              >
                9
              </span>
              <span
                class="day"
                title="10"
              >
                10
              </span>
              <span
                class="day selected"
                title="11"
              >
                11
              </span>
              <span
                class="day"
                title="12"
              >
                12
              </span>
              <span
                class="day"
                title="13"
              >
                13
              </span>
              <span
                class="day"
                title="14"
              >
                14
              </span>
              <span
                class="day"
                title="15"
              >
                15
              </span>
              <span
                class="day"
                title="16"
              >
                16
              </span>
              <span
                class="day"
                title="17"
              >
                17
              </span>
              <span
                class="day"
                title="18"
              >
                18
              </span>
              <span
                class="day"
                title="19"
              >
                19
              </span>
              <span
                class="day"
                title="20"
              >
                20
              </span>
              <span
                class="day"
                title="21"
              >
                21
              </span>
              <span
                class="day"
                title="22"
              >
                22
              </span>
              <span
                class="day"
                title="23"
              >
                23
              </span>
              <span
                class="day"
                title="24"
              >
                24
              </span>
              <span
                class="day"
                title="25"
              >
                25
              </span>
              <span
                class="day"
                title="26"
              >
                26
              </span>
              <span
                class="day"
                title="27"
              >
                27
              </span>
              <span
                class="day"
                title="28"
              >
                28
              </span>
              <span
                class="day"
                title="29"
              >
                29
              </span>
              <span
                class="day"
                title="30"
              >
                30
              </span>
            </div>
          </div>
        </div>
      </div>
    </body>

      33 |         <DatePicker onChange={mockHandleChange} value={june} weekDays={['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']} />
      34 |       );
    > 35 |       fireEvent.click(getByLabelText('previous month navigator'));
         |                       ^
      36 |
      37 |       expect(getByText('May 2019')).toBeInTheDocument();
      38 |     });

      at Object.getElementError (node_modules/@testing-library/dom/dist/config.js:30:19)
      at getAllByLabelText (node_modules/@testing-library/dom/dist/queries/label-text.js:121:38)
      at args (node_modules/@testing-library/dom/dist/query-helpers.js:54:17)
      at Object.<anonymous> (src/components/Inputs/DatePicker/DatePicker.spec.tsx:35:23)
```
### Problem description:

As far as I could see the query is just not working for span elements for some reason or maybe I'm being just super dumped right now which could also be the case. Then I apologize in advance ✊🐙

### Suggested solution:

For now I just used `getByTestId` which worked for me

**Comment by eps1lon** [comment]
You probably meant to 
```diff
- label="previous month navigator"
+ aria-label="previous month navigator"
```

It currently renders `<span label="" />` which is not a valid attribute for `<span />`. I could only find references of a `label` attribute for the `<track />` element.

**Comment by eps1lon** [comment]
Also: `getByTestId` queries `data-testid` not any labels. So you would need the above diff `getByLabel`.

But then again a label on a span isn't valid in the first place. I think you need to rethink your markup first.

**Comment by andrew-secret** [comment]
@eps1lon you are totally right if just mixed it up.. omg Thank you soo much for pointing this out to me ❤️
